public/app/core/components/LocalStorageWrapper/TimePickerHistory.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { PureComponent } from 'react';
+import { LocalStorageWrapper } from './';
+import { TimeRange, isDateTime } from '@grafana/data';
+
+const localStorageKey = 'grafana.dashboard.timepicker.history';
+
+interface Props {
+  children: (value: TimeRange[], onChange: (value: TimeRange) => void) => React.ReactNode;
+}
+
+export default class TimePickerWrapper extends PureComponent<Props> {
+  onAppendToHistory = (toAppend: TimeRange, values: TimeRange[], onSaveToStore: (values: TimeRange[]) => void) => {
+    if (!isAbsolute(toAppend)) {
+      return;
+    }
+    onSaveToStore(limit([...values, toAppend]));
+  };
+
+  render() {
+    const { children } = this.props;
+
+    return (
+      <LocalStorageWrapper<TimeRange[]> storeAtKey={localStorageKey} defaultValue={[]}>
```

Comment:
OK, so now a tricky thing. Should Explore and Dashboards share the same recent time ranges...? Can't see a reason not to, but who knows? @davkal any input on this?